### ISSUE TITLE
Enum Parameters Generate Invalid JSON

### ISSUE THREAD

**Issue by ghost** [adilbaig/gelfd#3]
I have a program which uses a lot of enum included in log messages. When sent to a graylog server, it cannot parse the JSON because the enum messages are not quoted.

For Example:
```
 enum SomeEnum{
	SE_A,
	SE_B,
	SE_C
}
auto m = gelf.Message("host", "A Message")
					.level(gelf.Level.WARNING)
					.timestamp(Clock.currTime())
					.anenum(myenum);
	
	
stdout.writeln(m); 
```

Results in the output: `{"version":1.1, "host":"host", "short_message":"A Message", "timestamp":1490636035, "level":4, "_anenum":SE_A}`

I corrected this locally by adding a static check for enum types in my code base.
protcol.d
```
import std.traits;
...
struct Field
    {
        bool enclose = false;
        string val;
        this(V)(V val)
        {
			static if (!is(V : size_t) || is(V == enum))
				this.enclose = true;
			
			this.val = to!string(val);
        }
    }
```

which fixes the problem as intended `{"version":1.1, "host":"host", "short_message":"A Message", "timestamp":1490637666, "level":4, "_anenum":"SE_A"}`

Thanks for your work,
Jeffrey Santi

**Comment by adilbaig** [comment]
Hey @jeffreysanti , 

Thanks for reporting this. Your proposed fix looked good to me, so I've merged it in and made a new release ([v1.3.3](https://github.com/adilbaig/gelfd/releases/tag/v1.3.3)). Let me know if there are issues.

Adil

**Comment by adilbaig** [comment]
Closing this card for now. Reopen if the issue persists
